base/client.jl

```diff
@@ -324,7 +324,7 @@ function load_julia_startup()
     else
         include_ifexists(Main, abspath(BINDIR, "..", "etc", "julia", "startup.jl"))
     end
-    include_ifexists(Main, abspath(homedir(), ".julia", "config", "startup.jl"))
+    include_ifexists(Main, abspath(DEPOT_PATH[1], "config", "startup.jl"))
```

Comment:
This shouldn't error if `DEPOT_PATH` is empty, so maybe
```suggestion
    !isempty(DEPOT_PATH) && include_ifexists(Main, abspath(DEPOT_PATH[1], "config", "startup.jl"))
```

Reply:
yes, defensive programming for sure. (I think the only way to get an empty DEPOT_PATH is to very explicitly set it that way)